samples/javascript_nodejs/14.nlp-with-dispatch/bots/dispatchBot.js

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityHandler } = require('botbuilder');
+const { LuisRecognizer } = require('botbuilder-ai');
+
+class DispatchBot extends ActivityHandler {
+    /**
+     * 
+     * @param {BotServices} botServices 
+     * @param {any} logger object for logging events, defaults to console if none is provided
+     */
+    constructor(dispatchRecognizer, qnaMaker, logger) {
+        super();
+        if (!dispatchRecognizer) throw new Error('[DispatchBot]: Missing parameter. dispatchRecognizer is required');
+        if (!qnaMaker) throw new Error('[DispatchBot]: Missing parameter. qnaMaker is required');
+        if (!logger) {
+            logger = console;
+            logger.log('[DispatchBot]: logger not passed in, defaulting to console');
+        }
+
+        this.logger = logger;
+        this.dispatchRecognizer = dispatchRecognizer;
+        this.qnaMaker = qnaMaker;
+
+        this.onMessage(async (context, next) => {
+            this.logger.log('Processing Message Activity.');
+
+            // First, we use the dispatch model to determine which cognitive service (LUIS or QnA) to use.
+            const recognizerResult = await dispatchRecognizer.recognize(context);
+
+            // Top intent tell us which cognitive service to use.
+            const topIntent = LuisRecognizer.topIntent(recognizerResult);
+
+            // Next, we call the dispatcher with the top intent.
+            await this.dispatchToTopIntentAsync(context, topIntent, recognizerResult);
```

Comment:
don't you need to call await next(); as the last call of your onMessage function?

Reply:
...YES...